keps/sig-apps/2232-jobs-stopped/README.md

```diff
@@ -0,0 +1,545 @@
+- Allow indefinitely delaying the creation of Pods owned by a Job.
+
+### Non-Goals
+
+- Being able to stop and resume jobs makes Job preemption, all-or-nothing
+  scheduling, and Job queueing possible. However, we don't propose to create
+  such higher-level controllers as a part of this KEP.
+- It might be useful to restrict who can set/modify the `stopped` flag.
+  However, we don't intend to create the validating and/or mutating webhooks
+  necessary to achieve that as a part of this KEP.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Consider a cloud provider where servers are cheaper in the night. My Job takes
+several days to complete, so I'd like to stop my Job early in the morning every
+day and resume it after dark to save money. However, I don't want to delete and
+re-create my Job every day as I don't want to lose track of completed Pods,
+logs, and other metadata.
+
+#### Story 2
+
+Let's say I'm a system administrator and there are many users submitting Jobs
+to my cluster. All user-submitted Jobs are created with `stopped: true`.
+There's only a finite amount of resources, so I must unpause these stopped Jobs
+in the right order at the right time.
+
+I can write a higher-level Job queueing controller to do this based on external
+factors. For example, the controller could choose to simply unpause Jobs in the
+FIFO order. Alternatively, Jobs could be assigned priority and, just like
+kube-scheduler, the controller can make a decision based on the stopped Job
+queue (including Job preemption). Each Job could request a different amount of
+resources, so the higher-level controller may also want to resize the cluster
+to just fit the Job it's going to run. Regardless of what logic the controller
+uses to queue jobs, being able to stop Jobs indefinitely and then resume them
+later is important.
+
+### Notes/Constraints/Caveats
+
+System administrators may want to restrict who can set/modify the `stopped`
+field when creating or updating Jobs. This can be achieved with validating
+and/or mutating webhooks.
+
+When a Job is stopped, the time is recorded as part of the Job status.  Users
+can infer how long a Job has been paused for through this field. This can be
+useful when making decisions around which Job should be resumed.
+
+A Job that is complete cannot be stopped.
+
+### Risks and Mitigations
+
+1. Stopping an active Job deletes *all* active pods. Users must design their
+   application to gracefully handle this.
+
+## Design Details
+
+We propose adding a `stopped` field to the JobSpec API:
+
+```go
+type JobSpec struct {
+	// Stopped specifies whether the Job controller should create Pods or not. If
+	// a Job is created in the stopped state, no Pods are created by the Job
+	// controller. If a Job enters the stopped state (i.e. the flag goes from
+	// false to true), the Job controller will delete all Pods associated with
+	// this Job. This is an alpha field and requires enabling the StoppedJob
```

Comment:
controller will delete all **running** Pods, not **all** Pods, which is what you said elsewhere.

Reply:
I'd also mention here explicitly that it is user's responsibility to design an application such that it can be suspended half way through, which means some of the pods will be terminated.